apis/supervisor/config/v1alpha1/types_oidcclient.go.tmpl

```diff
@@ -75,17 +75,20 @@ type OIDCClientSpec struct {
 
 // OIDCClientStatus is a struct that describes the actual state of an OIDCClient.
 type OIDCClientStatus struct {
-	// Phase summarizes the overall status of the OIDCClient.
+	// phase summarizes the overall status of the OIDCClient.
 	// +kubebuilder:default=Pending
 	// +kubebuilder:validation:Enum=Pending;Ready;Error
 	Phase OIDCClientPhase `json:"phase,omitempty"`
 
-	// Represents the observations of an OIDCClient's current state.
+	// conditions represent the observations of an OIDCClient's current state.
 	// +patchMergeKey=type
 	// +patchStrategy=merge
 	// +listType=map
 	// +listMapKey=type
 	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// totalClientSecrets is the current number of client secrets that are detected for this OIDCClient.
+	TotalClientSecrets int `json:"totalClientSecrets"`
```

Comment:
Didn't know that! Thanks. Updated in be85e1ed0a720f784d542ea8819cab9156a1b52e.

I used a simple cast in the controller which is probably fine because it is never going to find more than 2,147,483,647 actual client secrets stored in the Kube `Secret`, so it should never be able to overflow in practice.